### ISSUE TITLE
False positive for UnnecessaryParentheses in require + in

### ISSUE THREAD

**Issue by nvkleban** [detekt/detekt#7351]
## Observed Behavior

The following snippet triggers the UnnecessaryParentheses rule:

```
val progressBytes: Int
val totalBytes: Int
require(progressBytes in 0..totalBytes)
```

## Expected Behavior

UnnecessaryParentheses should not be triggered as there is no way to get rid of brackets.

## Your Environment

* Version of detekt used: 1.23.6
* Version of Gradle used: 8.6
* Kotlin: 2.0
